### DISCUSSION TITLE
Leave modifier for x-intersect

### DISCUSSION THREAD

**Discussion by SimoTod** [alpinejs/alpine#1921] [3. Feature ideas]
Chatting to other users on the discord channel, we feel that a leave modifier for x-intersect would be really useful and it's been asked a couple of times.

For example, a page with sections and a fixed menu with anchor links to jump at the right point of the page. Using x-intersect, we are able to apply a special class to the menu item to "highlight" it but we don't have a callback to remove that class when we leave the section.

Would you accept a PR for this or do you think there's no enough request to justify it? Thanks.

**Comment by MuzafferDede** [comment]
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Is there any reason &quot;.out/.leave&quot; modifiers are gone? Thinking about a PR for that with a different approach if it&#39;s still on the roadmap.</p>&mdash; Muzaffer Dede (@MuzafferDede) <a href="https://twitter.com/MuzafferDede/status/1414896753720004609?ref_src=twsrc%5Etfw">July 13, 2021</a></blockquote> 

I did mention this while ago, didn't got a response. Shouldn't be hard to achieve.

**Reply by Pronian** [reply]
I also think that it would be useful

**Reply by MuzafferDede** [reply]
Check this PR #1928 , it think we got it!

**Reply by SimoTod** [reply]
> Check this PR #1928 , it think we got it! 

The only thing is that the PR is a breaking change. People that are using the plugin right now will get their website broken (let's be honest, very few read the release notes, hahaha).

Let's wait for Caleb but you may have to use a different syntax (x-intersect:leave or x-intersect.leave).

Also, alpine uses browser native events, it doesn't triggers its own so it could confuse people. Should they have a namespace, `change` conflicts with a native event, if you put that on an input tag or a select, you can get unexpected behaviours.

**Reply by MuzafferDede** [reply]
Breaking change? There is no any breaking change, correct me if i am wrong. 

I think you haven't seen the update i made which i change the name "change" to "changed". 

Not sure if you really got the idea. Feels to me your mind is set to modifier based solution but yeah, i think event base solution could be a starting point for many use cases.

**Reply by SimoTod** [reply]
My bad, I was confused by the event part and I didn't see you kept the old code too.

Just ignore my comment, I'm not the one approving PRs :)

**Reply by MuzafferDede** [reply]
No worries, after i considered your comment, actually i think could create another version of it based on directive value as `x-intersect:leave` `x-intersect:enter`. Should i create that one as well? That might be more like "Alpine" way and Caleb might like as well.

**Reply by SimoTod** [reply]
I would wait for Caleb to have a look, maybe just ask on the PR if he prefer the other approach and you'll update it after. Sending 2 different PRs for the same feature is going to make it more confusing for Caleb to review.

**Comment by MuzafferDede** [comment]
- #1933 got merged! We have `x-intersect:enter` and `x-intersect:leave` in next release!
